### DISCUSSION TITLE
What is the "sort different domains independently" setting in open & share module?

### DISCUSSION THREAD

**Discussion by douchesfeedboard** [TrianguloY/URLCheck#199] [Q&A] [answered]
Hello,

As written in the title, I don't understand what is the setting "sort different domains independently" in the "open & share module".
Can anyone explain to me what this setting can do?

Thank you.

**Comment by TrianguloY** [comment, marked as answer]
Any time you open a link with a specific app, it is saved as "preferred" using a simple algorithm of "the more you choose an app, the more it is preferred". This will make that app appear sooner than the others, and ultimately at first position (the main big button).
Imagine you have three apps that can open a link, A B C. At first it'll use whatever android suggests, for example ABC. If you choose C the app will remember "C is preferred over A and B" so next time it'll show CAB.

If that is disabled this process apply to all urls, so no matter which link you open C will always be before A and B.
If the flag is enabled the preferred will differentiate depending on the link domain, so for example if you choose C when opening a google.com link, next google.com link will do show CAB but a github.com will still show ABC as it's a different domain.

In the end it's just a matter of whether the android default order is good for you or not. I have the setting disabled because android keeps suggesting chrome over Firefox (even though I almost never use chrome) so I want all links to remember that preference. Other users are happy with the default, but maybe they want only twitter links to open in Opera, so they may enable it.

I know it's a very basic and no configurable algorithm, a new more powerful one may be included somewhere in the future. But for now, that's what it is.



You can find some more info here:
 https://github.com/TrianguloY/UrlChecker/issues/192#issuecomment-1444570104 https://github.com/TrianguloY/UrlChecker/issues/111

**Comment by douchesfeedboard** [comment]
Thank you for the answer!
